streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetadataState.java

```diff
@@ -50,7 +51,7 @@
     private final HostInfo thisHost;
     private List<StreamsMetadata> allMetadata = Collections.emptyList();
     private Cluster clusterMetadata;
-    private StreamsMetadata localMetadata;
+    private AtomicReference<StreamsMetadata> localMetadata;
```

Comment:
We need to actually initialize this now, although we can just initialize the value to null. ie
```
private AtomicReference<StreamsMetadata> localMetadata = new AtomicReference<>(null);
```